MobileBuy/buy/src/main/java/com/shopify/buy3/GraphCall.java

```diff
@@ -30,36 +30,129 @@
+   * Creates a new, identical call to this one which can be enqueued or executed even if this call has already been executed or canceled.
+   *
+   * @return new cloned call
+   */
   @NonNull GraphCall<T> clone();
 
+  /**
+   * Sends the {@code GraphQL} request immediately and blocks until the response can be processed or is an error.
+   *
+   * @return {@code GraphQL} response
+   * @throws IllegalStateException when the call has already been executed
+   * @throws GraphError            if the request could not be executed due to a cancellation, a timeout, network failure, parsing error
+   *                               etc.
+   * @see GraphError
+   */
   @NonNull GraphResponse<T> execute() throws GraphError;
 
+  /**
+   * Schedules the call to be executed at some point in the future.
+   *
+   * @param callback which will handle the response or a failure exceptions.
+   * @return call that has been scheduled for execution
+   * @throws IllegalStateException when the call has already been executed
+   * @see Callback
+   */
   @NonNull GraphCall<T> enqueue(@NonNull Callback<T> callback);
 
-  @NonNull GraphCall<T> enqueue(@NonNull Callback<T> callback, @Nullable Handler handler);
-
+  /**
+   * Schedules the call to be executed at some point in the future.
+   *
+   * @param callback        which will handle the response or a failure exceptions.
+   * @param callbackHandler the callback will be run on the thread to which this handler is attached to
+   * @return call that has been scheduled for execution
+   * @throws IllegalStateException when the call has already been executed
+   * @see Callback
+   */
+  @NonNull GraphCall<T> enqueue(@NonNull Callback<T> callback, @Nullable Handler callbackHandler);
+
+  /**
+   * Schedules the call to be executed at some point in the future.
+   *
+   * @param callback     which will handle the response or a failure exceptions.
+   * @param handler      the callback will be run on the thread to which this handler is attached to
+   * @param retryHandler request retry policy
+   * @return call that has been scheduled for execution
+   * @throws IllegalStateException when the call has already been executed
+   * @see Callback
+   * @see RetryHandler
+   */
   @NonNull GraphCall<T> enqueue(@NonNull Callback<T> callback, @Nullable Handler handler, @NonNull RetryHandler retryHandler);
 
+  /**
+   * Callback used for notifying response of GraphQL call that has been scheduled for execution.
+   *
+   * @param <T> type of the {@link AbstractResponse} response this call back serves
+   */
   abstract class Callback<T extends AbstractResponse<T>> {
+
+    /**
+     * Called when GraphQL response is received and parsed successfully.
+     *
+     * @param response parsed GraphQL response
+     */
     public abstract void onResponse(@NonNull GraphResponse<T> response);
 
+    /**
+     * Called when GraphQL call can't be executed due to cancellation, a connectivity problem, timeouts, parsing errors etc.
```

Comment:
Same issue. It seems weird that `onFailure()` would be called if the app called `cancel()` on the request